File: web_admin/src/domain/repositories/PayslipRepository.ts
import type { Payslip } from '@/domain/hr/types';
import type { Unsubscribe } from './AuthRepository';

export interface PayslipRepository {
  watchAll(cb: (payslips: Payslip[]) => void): Unsubscribe;
  getById(id: string): Promise<Payslip | null>;
  create(input: Omit<Payslip, 'id' | 'createdAt'>): Promise<string>;
  delete(id: string): Promise<void>;
}

File: lib/domain/repositories/payslip_repository.dart
import 'package:maki_mobile_pos/domain/entities/entities.dart';

/// Generated payslips — frozen snapshots (firestore.rules: update:false).
/// Deliberately no update method: a slip is created, read, or deleted.
abstract class PayslipRepository {
  /// Streams all payslips, periodStart DESC then employeeName ASC
  /// (web's fixed sort).
  Stream<List<PayslipEntity>> watchAll();

  Future<PayslipEntity?> getById(String id);

  /// Creates the frozen snapshot; returns the new doc id.
  Future<String> create(PayslipEntity payslip);

  Future<void> delete(String id);
}
